.tools/readmes/render.py

```diff
@@ -306,9 +306,12 @@ def write(self):
             f.write(self.readme_text)
 
     def check(self):
-        with self.readme_filename.open("r", encoding="utf-8") as f:
-            readme_current = f.read()
-            return readme_current == self.readme_text
+        try:
+            with self.readme_filename.open("r", encoding="utf-8") as f:
+                readme_current = f.read()
+                return readme_current == self.readme_text
+        except FileNotFoundError:
+            return False
```

Comment:
Don't you want `return True`? So that the caller will write the new output? Otherwise I think this will put the new file in the "unchanged" list without writing it.

Reply:
Just as a point of reference, I already fixed this several weeks ago but David's refactor overwrote the changes and undid the fix. Here is the PR that shows similar logic implemented in IMO a more elegant way: https://github.com/awsdocs/aws-doc-sdk-examples/pull/6592/files